operators/src/pro/cache/util.rs

```diff
@@ -0,0 +1,224 @@
+use geoengine_datatypes::util::ByteSize;
+use snafu::ensure;
+
+use super::error::CacheError;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct CacheSize {
+    byte_size_total: usize,
+    byte_size_used: usize,
+}
+
+impl CacheSize {
+    pub fn new(byte_size_total: usize) -> Self {
+        Self {
+            byte_size_total,
+            byte_size_used: 0,
+        }
+    }
+
+    #[inline]
+    pub fn total_byte_size(&self) -> usize {
+        self.byte_size_total
+    }
+
+    #[inline]
+    pub fn byte_size_used(&self) -> usize {
+        self.byte_size_used
+    }
+
+    #[inline]
+    pub fn byte_size_free(&self) -> usize {
+        if self.byte_size_used > self.byte_size_total {
+            0
+        } else {
+            self.byte_size_total - self.byte_size_used
+        }
+    }
+
+    #[inline]
+    pub fn size_used_fraction(&self) -> f64 {
+        self.byte_size_used as f64 / self.byte_size_total as f64
+    }
+
+    #[inline]
+    pub fn can_fit_bytes(&self, bytes: usize) -> bool {
+        !self.is_overflowing() && self.byte_size_free() >= bytes
```

Comment:
isn't `!self.is_overflowing()` redundant here?

Reply:
ok, I see, it's because you also check it with bytes == 0, right?

Reply:
yes its a bit strange but the capacity is negative in this case. However we can't really use negative capacity with usizes,